### PR TITLE
configprofiles,autoconfig: remove autoconfig

### PR DESCRIPTION
**Pull request by dt**
First commit is #119315.

Release note (breaking change): the --config-profile command line flag added for preview support of PCR
in the previous release has been removed. Nodes, including those for clusters on which PCR may be used,
are instead started normally, with the one-time steps to configure the cluster for PCR being performed
via SQL statements instead. Specifically, to use PCR, on the source cluster one must 1) create an application tenant, 2) switch it to be the default, and 3) enable rangefeeds.

Epic: none.

### CODE REVIEW

**Review by stevendanna — COMMENTED** [review]
This alone

Screenshot 2024-02-20 at 09 32 58

Makes a pretty strong argument.  We might want to list in the commit message some of the things we found suboptimal about the config-profiles approach:

- Configuration profiles were applied once and were opaque to the user (and support);
- They were not clearly "reversible" because of that opaqueness; 
- They were not symmetric; and
- They relied on a relatively complex, generalised implementation that we don't currently have plans to make use of.

If we were to, say, enable Rangefeeds for secondary tenants and always create a secondary tenant (or provided some way at tenant creation to set a tenant as the default), then we really could be down to a single SQL command to enable "UA" mode.

I've left some very minor nits.  I believe the plan is to chat through this at the next Friends of MT meeting.

File: pkg/jobs/retired.go [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+
+func noopConstructor(j *Job, _ *cluster.Settings) Resumer {
+	payload := j.Payload()
+	return noopResumer(fmt.Sprintf("%d %s", j.ID(), payload.Type()))
+}
+
+type noopResumer string
+
+func (r noopResumer) Resume(ctx context.Context, _ interface{}) error {
+	log.Infof(ctx, "completing retired job %s", r)
```
[the remark this answers is not in the document]
Comment (dt) [reply pkg/jobs/retired.go:43]: We could. Or we could delete the logline instead -- do we think it is useful?

[the remark this answers is not in the document]
Comment (stevendanna) [reply pkg/jobs/retired.go:43]: I don't think it is that useful.  The job system will log these jobs entering the new state, along with the type and the ID. Hopefully the list of retired jobs never grows large enough that we'd be confused about whether a particular job was retired or not.

[the remark this answers is not in the document]
Comment (dt) [reply pkg/jobs/retired.go:43]: Removed.

File: pkg/cli/cliflags/flags.go [hunk 2]
```diff
@@ -1445,12 +1445,6 @@ Disable the creation of a default dataset in the demo shell.
 This makes 'cockroach demo' faster to start.`,
 	}
 
-	ConfigProfile = FlagInfo{
-		Name:        "config-profile",
-		EnvVar:      "COCKROACH_CONFIG_PROFILE",
-		Description: `Select a configuration profile to apply.`,
-	}
-
```
[the remark this answers is not in the document]
Comment (dt) [reply pkg/cli/cliflags/flags.go:1453]: Added to the commit message.

**Comment by dt** [comment]
> list in the commit message some of the things we found suboptimal about the config-profiles
Done.

**Comment by dt** [comment]
> believe the plan is to chat through this at the next Friends of MT meeting.

I interpreted that suggestion as a retrospective review of the stack of changes that have happened over the last week, rather than just this PR, or blocking this for a couple weeks until we can meet?

**Review by stevendanna — APPROVED** [review]
Overall I'm on board with this.

**Comment by dt** [comment]
TFTR!

bors r+

**Pull request merged**